README.md

```diff
@@ -0,0 +1,238 @@
+    .fetch_one(executor)
+    .await
+    .map_err(|e| crate::Error::DbError(e.into()))?;
+
+    Ok(Task {
+        id: record.id,
+        description: task.description,
+    })
+}
+```
+
+The crate uses a simple folder structure:
+
+```
+db
+├── migrations // Database migrations as plain SQL files
+├── src
+    ├── entities     // Entity structs, changesets and related functions for retrieving and persisting records (see example above)
+    └── test-helpers // Functions for retrieving and persisting records that are only relevant for tests (these are defined behind the `test-helpers` feature)
+```
+
+### The `config` crate
+
+The `config` crate contains the struct that holds all configuration values at runtime as well as code for parsing the configuration based on a hierarchy of TOML files and environment variables. The `Config` struct contains fields for the server and database configuration and can be extended freely:
+
+```rs
+#[derive(Deserialize, Clone, Debug)]
+pub struct Config {
+    pub server: ServerConfig,
+    pub database: DatabaseConfig, // The database configuration only exists for projects that use a database
+    // add your config settings here…
+}
+```
+
+The values for the server and database configuration are read from the `APP_SERVER__IP`, `APP_SERVER__PORT`, and `APP_DATABASE__URL` environment variables. Any application-specific settings are first read from `app.toml` and then from an environment-specific file, e.g. `production.toml` so that environment-specific settings override those in `app.toml`. The main fils and folders in the crate are:
+
+```
+config
+├── environments
+|   ├── development.toml // Configuration settings specific for the development environment
+|   ├── production.toml  // Configuration settings specific for the production environment
+|   └── test.toml        // Configuration settings specific for the test environment
+├── src
+|   └── lib.rs           // Contains the `Config` struct and code for constructing it based on the configuration files and environment variables
+└── app.toml             // Basis configuration settings that will be overridden by the same settings in the respective environment-specific configuration file
+```
+
+### The `cli` crate
+
+The `cli` crate contains the `db` (which only exists for projects that use a database) and `generate` binaries for running database operations such as creating or dropping the database or running migrations, and generating project files such as entities, controllers, or middlewares. The workspace is configured so that those binaries can be run with just `cargo db` and `cargo generate`:
+
+```
+» cargo db
+A CLI tool to manage the project's database.
+
+Usage: db [OPTIONS] <COMMAND>
+
+Commands:
+  drop     Drop the database
+  create   Create the database
+  migrate  Migrate the database
+  reset    Reset (drop, create, migrate) the database
+  seed     Seed the database
+  help     Print this message or the help of the given subcommand(s)
```

Comment:
It'd be awesome to include those in the generated Readme if possible.

```suggestion
Commands:
  drop     Drop the database
  create   Create the database
  migrate  Migrate the database
  reset    Reset (drop, create, migrate) the database
  seed     Seed the database
  help     Print this message or the help of the given subcommand(s)
```

Reply:
The READMEs for the generated projects are to be done still